src/playlist-loader.js

```diff
@@ -268,11 +330,60 @@ export default class PlaylistLoader extends EventTarget {
         // only refresh the media playlist if no other activity is going on
         return;
       }
+      const media = this.media();
+
+      let uri = resolveUrl(this.master.uri, media.uri);
+
+      if (this.experimentalLLHLS && !media.endList) {
+        const query = [];
```

Comment:
The constructor is getting a bit long. May be worth breaking this into its own function. Can even be outside  of the class and return an object with necessary details to make testing easier.

Reply:
I moved mediaupdatetimeout's handler to a new function and that made it much better.